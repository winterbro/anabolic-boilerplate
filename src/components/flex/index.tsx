import { FlexBox, FlexBoxProps } from 'components/flex/FlexBox';
import {
  FlexContainer,
  FlexContainerProps,
} from 'components/flex/FlexContainer';

export type { FlexBoxProps, FlexContainerProps };
export { FlexBox, FlexContainer };
