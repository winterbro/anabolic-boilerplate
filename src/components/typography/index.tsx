import Heading from 'components/typography/Heading';

export default Heading;
