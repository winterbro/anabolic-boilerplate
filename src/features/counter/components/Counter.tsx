import { useAppDispatch, useAppSelector } from 'store/hooks';
import { counterSlice } from 'store';
import styled from 'styled-components';

const StyledCounter = styled.button`
  padding: 0.6em 1.2em;
  margin-bottom: var(--space-xs);
  font-family: inherit;
  color: #fff;
  cursor: pointer;
  background-color: #1a1a1a;
  border: 1px solid transparent;
  border-radius: 8px;
  transition: border - color 0.25s;

  &:hover {
    border-color: #646cff;
  }

  &:focus,
  &:focus-visible {
    outline: 4px auto - webkit - focus - ring - color;
  }
`;

function Counter() {
  const count = useAppSelector((state) => state.counter);
  const dispatch = useAppDispatch();

  return (
    <StyledCounter
      type="button"
      onClick={() => dispatch(counterSlice.actions.increment())}
    >
      count is {count.value}
    </StyledCounter>
  );
}

export { Counter, StyledCounter };
