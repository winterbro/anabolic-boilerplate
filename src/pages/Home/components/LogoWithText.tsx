import styled from 'styled-components';

export type LogoProps = {
  linkSrc: string;
  imageSrc: string;
  text: string;
  color?: string;
};

const StyledLogoLink = styled.a<Pick<LogoProps, 'color'>>`
  display: inline-flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  color: #888;
  text-decoration: inherit;

  &:hover {
    color: ${(props) => props.color ?? '#646cffaa'};
  }
`;

const StyledLogo = styled.img<Pick<LogoProps, 'color'>>`
  height: 4em;
  margin: 1em;

  &:hover {
    filter: drop-shadow(0 0 1em ${(props) => props.color ?? '#646cffaa'});
  }
`;

function LogoWithText({ linkSrc, imageSrc, text, color }: LogoProps) {
  return (
    <StyledLogoLink
      href={linkSrc}
      color={color}
      target="_blank"
      rel="noreferrer"
    >
      <StyledLogo src={imageSrc} color={color} alt={`${text} Logo`} />
      {text}
    </StyledLogoLink>
  );
}

export { StyledLogo, LogoWithText };
