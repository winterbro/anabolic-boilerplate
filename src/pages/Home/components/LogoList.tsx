import logos from 'pages/Home/data/logos.json';
import { FlexBox } from 'components/flex';
import { LogoWithText } from './LogoWithText';

function Logos() {
  return (
    <FlexBox gap="var(--space-sm)">
      {logos !== null &&
        logos.map((logo) => {
          return (
            <LogoWithText
              key={logo.name}
              linkSrc={logo.url}
              imageSrc={logo.image}
              text={logo.name}
              color={logo.color}
            />
          );
        })}
    </FlexBox>
  );
}

export default Logos;
