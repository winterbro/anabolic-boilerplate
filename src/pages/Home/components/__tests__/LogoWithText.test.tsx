import { describe, it } from 'vitest';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { LogoWithText, StyledLogo } from 'pages/Home/components/LogoWithText';
import viteLogo from 'assets/vite.svg';

describe('LogoWithText', () => {
  it('should render a logo with text', () => {
    render(
      <LogoWithText
        linkSrc="https://vite.dev"
        imageSrc={viteLogo}
        text="Vite"
      />
    );
    const logo = screen.getByRole('img');
    const logoText = screen.getByText('Vite');
    expect(logo).toBeInTheDocument();
    expect(logoText).toBeInTheDocument();
  });
});

describe('StyledLogo', () => {
  it('should render the styled component correctly', () => {
    const tree = renderer.create(<StyledLogo />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
