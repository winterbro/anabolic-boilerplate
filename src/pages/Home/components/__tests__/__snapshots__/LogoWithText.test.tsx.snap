// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StyledLogo > should render the styled component correctly 1`] = `
.c0 {
  height: 4em;
  margin: 1em;
}

.c0:hover {
  -webkit-filter: drop-shadow(0 0 1em #646cffaa);
  filter: drop-shadow(0 0 1em #646cffaa);
}

<img
  className="c0"
/>
`;
